src/platformsh.js

```diff
@@ -1,67 +1,447 @@
+                    for (let [url, route] of Object.entries(this.routesDef)) {
+                        route['url'] = url;
+                    }
+                }
+
+                let relationships = this._getValue('RELATIONSHIPS');
+                if (relationships) {
+                    this.relationshipsDef = decode(relationships);
+                }
+            }
+
+            let variables = this._getValue('VARIABLES');
+            if (variables) {
+                this.variablesDef = decode(variables);
+            }
+
+            let application = this._getValue('APPLICATION');
+            if (application) {
+                this.applicationDef = decode(application);
+            }
+        }
+
+
+    }
+
+    /**
+     * Checks whether the code is running on a platform with valid environment variables.
+     *
+     * @return {boolean}
+     *   True if configuration can be used, false otherwise.
+     */
+    isValidPlatform() {
+        return Boolean(this._getValue('APPLICATION_NAME'));
+    }
+
+    /**
+     * Checks whether the code is running in a build environment.
+     *
+     * If false, it's running at deploy time.
+     *
+     * @return {boolean}
+     */
+    inBuild() {
+        return this.isValidPlatform() && !Boolean(this._getValue('ENVIRONMENT'));
+    }
+
+    /**
+     * Checks whether the code is running in a runtime environment.
+     *
+     * @return {boolean}
+     */
+    inRuntime() {
+        return this.isValidPlatform() && Boolean(this._getValue('ENVIRONMENT'));
+    }
+
+    /**
+     * Determines if the current environment is a Platform.sh Enterprise environment.
+     *
+     * @return {boolean}
+     *   True on an Enterprise environment, False otherwise.
+     */
+    onEnterprise() {
+        return this.isValidPlatform() && this._getValue('MODE') == 'enterprise';
+    }
+
+    /**
+     * Determines if the current environment is a production environment.
+     *
+     * Note: There may be a few edge cases where this is not entirely correct on Enterprise,
+     * if the production branch is not named `production`.  In that case you'll need to use
+     * your own logic.
+     *
+     * @return {boolean}
+     *   True if the environment is a production environment, false otherwise.
+     *   It will also return false if not running on Platform.sh or in the build phase.
+     */
+    onProduction() {
+        if (!this.inRuntime()) {
+            return;
+        }
+
+        let prodBranch = this.onEnterprise() ? 'production' : 'master';
+
+        return this._getValue('BRANCH') == prodBranch;
+    }
+
+    /**
+     * Returns the routes definition.
+     *
+     * @return {object}
+     *   The routes definition object.
+     * @throws {Error}
+     *   If the routes are not accessible due to being in the wrong environment.
+     */
+    routes() {
+        if (!this.isValidPlatform()) {
+            throw new Error('You are not running on Platform.sh, so routes are not available.');
```

Comment:
You probably want to subclass `Error` there.

Reply:
From the googling I did I wasn't sure what the standard practice was for JS.  Like, do you subclass Error, or do you just throw strings, because those are legit objects?  Most of the examples I found seemed to imply the latter.

Reply:
https://medium.com/@xjamundx/custom-javascript-errors-in-es6-aa891b173f87